pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotTaskManager.java

```diff
@@ -63,99 +65,147 @@ public PinotTaskManager(PinotHelixTaskResourceManager helixTaskResourceManager,
   }
 
   /**
-   * Returns the cluster info provider.
-   * <p>
-   * Cluster info provider might be useful when initializing task generators.
-   *
-   * @return Cluster info provider
+   * Returns the cluster info accessor.
+   * <p>Cluster info accessor can be used to initialize the task generator.
    */
   public ClusterInfoAccessor getClusterInfoAccessor() {
     return _clusterInfoAccessor;
   }
 
   /**
    * Registers a task generator.
-   * <p>
-   * This method can be used to plug in custom task generators.
-   *
-   * @param pinotTaskGenerator Task generator to be registered
+   * <p>This method can be used to plug in custom task generators.
    */
-  public void registerTaskGenerator(PinotTaskGenerator pinotTaskGenerator) {
-    _taskGeneratorRegistry.registerTaskGenerator(pinotTaskGenerator);
+  public void registerTaskGenerator(PinotTaskGenerator taskGenerator) {
+    _taskGeneratorRegistry.registerTaskGenerator(taskGenerator);
   }
 
   /**
-   * Public API to schedule tasks. It doesn't matter whether current pinot controller is leader.
+   * Public API to schedule tasks (all task types) for all tables. It might be called from the non-leader controller.
+   * Returns a map from the task type to the task scheduled.
    */
   public synchronized Map<String, String> scheduleTasks() {
-    Map<String, String> tasksScheduled = scheduleTasks(_pinotHelixResourceManager.getAllTables());
-
-    // Reset the task because this method will be called from the Rest API instead of the periodic task scheduler
-    // TODO: Clean up only the non-leader tables instead of all tables
-    cleanUpTask();
-    setUpTask();
-
-    return tasksScheduled;
+    return scheduleTasks(_pinotHelixResourceManager.getAllTables(), false);
```

Comment:
why this is always false?

Reply:
This method is connected to the rest endpoint, so the caller might not be the leader controller

Reply:
ic, so we actually always assume this call is from a non-leader.

Reply:
Yes, if it is not always the leader, we need to assume it is not the leader